hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -137,14 +142,11 @@ public void validateSemantics(
                 final var senderId = nftTransfer.orElseThrow().senderAccountIDOrThrow();
                 final var senderAccount = accountStore.getAliasedAccountById(senderId);
                 validateTrue(senderAccount != null, INVALID_ACCOUNT_ID);
-                validateNftTransfers(
-                        context.payer(),
-                        senderAccount,
-                        tokenId,
-                        xfers.nftTransfers(),
-                        tokenRelStore,
-                        tokenStore,
-                        nftStore);
+                validateNftTransfers(senderAccount, tokenId, xfers.nftTransfers(), tokenRelStore, tokenStore, nftStore);
+
+                totalNftTransfers += xfers.nftTransfers().size();
+                // Verify that the current total number of (counted) nft transfers does not exceed the limit
+                validateTrue(totalNftTransfers <= legerConfig.nftTransfersMaxLen(), BATCH_SIZE_LIMIT_EXCEEDED);
```

Comment:
Curious, why use `BATCH_SIZE_LIMIT_EXCEEDED` as the error here? I would've expected `TOKEN_TRANSFER_LIST_SIZE_LIMIT_EXCEEDED` (again)

Reply:
I'm wondering about the same thing, but since this is the response code returned from the crypto transfer, I decided to use it here too.